src/user/user.service.ts

```diff
@@ -126,36 +125,86 @@ export class UserService implements OnModuleInit {
     return await this.userDatabaseService.updateUser(id, updateParams)
   }
 
-  // TODO: - Return reduce into updateUserRecentTodos for create and update methods (to prevent redundant access to the DB);
-  async updateUserRecentTodos(todo: Todo) {
-    const userId = todo.userId.toString()
+  async updateUserRecentTodosOnTodoCreate() {
+    for await (const createdTodo of this.todosChangeStreamDatabaseService.subscribeOnTodoCreate()) {
+      // ? Question: access DB for deprecated user.todos by `getUserById` to update them manually, or access DB for user.todos directly by the `getUserTodos` using query params? In both approaches we access DB, so why not the second one? In both options we access for the same data.
+
+      // Option 1:
+      const user = await this.getUserById(createdTodo.userId)
+
+      const recentUserTodos = user.todos.reduce(
+        (todos, todo) => {
+          if (todos.length < USER_RECENT_TODOS_COUNT) {
+            todos.push(todo)
+          }
+          return todos
+        },
+        [createdTodo],
+      )
 
-    const recentUserTodos = await this.getUserTodos(userId, {
-      sort: { updatedAt: -1 },
-      limit: USER_RECENT_TODOS_COUNT,
-    })
+      await this.userDatabaseService.updateUser(createdTodo.userId, {
+        todos: recentUserTodos,
+      })
+
+      // Option 2:
+      /*
+      const userId = recentTodo.userId.toString()
+
+      const recentUserTodos = await this.getUserTodos(userId, {
+        sort: { updatedAt: -1 },
+        limit: USER_RECENT_TODOS_COUNT,
+      })
+  
+      return await this.userDatabaseService.updateUser(userId, {
+        todos: recentUserTodos,
+      })
+      */
+    }
+  }
```

Comment:
Just highlighting of this comment:
// ? Question: access DB for deprecated user.todos by `getUserById` to update them manually, or access DB for user.todos directly by the `getUserTodos` using query params? In both approaches we access DB, so why not the second one? In both options we access for the same data.

We already discussed it before, but as I remebmer, the main goal of returning the reduce method is an accessing the DB less times. But in both of these options we access the DB at least 1 time. So, the Option 2 looks quite sufficient here, isn't it?